### DISCUSSION TITLE
Support for other languages

### DISCUSSION THREAD

**Discussion by bluenyx** [Shelf-nu/shelf.nu#221] [Ideas]
Will you support language files that can be translated?

**Comment by carlosvirreira** [comment]
Hi @bluenyx - we have not considered this nor it is on the immediate roadmap. I am curious regardless, to which language would you like to translate shelf? How would you approach it?

**Comment by DonKoko** [comment]
Hey @bluenyx. I have transformed your issue to a discussion as we would like to see if there are more people interested. Overall its not out of the question to implement localization using something like i18n and community based translations. 
Lets see if there are more people interested and we can consider it.

**Reply by Pablohn26** [reply]
Please consider adding multi language support as this tool could be pretty useful in developing countries

**Comment by L30N3N** [comment]
Hi, 

I am also interested in a translation of Shelf. My suggestions in this context would be the following:

1. analyze the user groups of Shelf, which countries do the current users come from?
2. which countries would you like to address in particular?
3. which languages cover a large part of the world? (e.g. https://en.wikipedia.org/wiki/List_of_languages_by_number_of_native_speakers)

Now take the findings from these checks and internationalize the software, concentrating on the most important 5-10 languages in the first step. 

I would offer the whole thing as a beta feature for the time being, whereby the individual user can select their preferred language in the settings. 

New features will initially only be available in the standard language (English) until a translation is available.

For the translation, I would rely on the community and tools such as Crowdin and one "expert" per language who accepts the translations.

Also, if possible, automate the whole thing so that there is little work for the developers and the whole thing works on the side and offers added value for the users. 

Translated with DeepL.com (free version)

**Reply by DonKoko** [reply]
Thanks for your suggestion and detailing how to handle this. This is something that will definitely come to shelf however we have to discuss it further with the team, as it will take lots of effort and time to implement. We currently have a couple of thousand users and we have only had the request a few times so far so its very hard to justify having to spend lots of extra time for each feature.

This being said, as you know this is an open source project so suggestions and PR's are always welcome. If someone is willing to contribute then we will be totally willing to collaborate to make this a reality.

Just curious, what language are you interested in?

**Reply by L30N3N** [reply]
I am interested in the translation into German. And I am also in the process of contributing things to the development.

**Reply by DonKoko** [reply]
We will consider this further and place our updates on this discussion, once we have something to share.

**Comment by Matyu9** [comment]
Hi,
Is there any update on this subject? 
Thanks,

**Reply by DonKoko** [reply]
We had someone working on a PR for this but I think they never finished it. Feel free to take a look at the PR and give it a go if interested.
